packages/builder/src/extrinsic/mrl/wormhole/ethereumXcm/ethereumXcm.ts

```diff
@@ -0,0 +1,122 @@
+      isAutomatic: boolean;
+    }): MrlExtrinsicConfigBuilder => ({
+      build: ({
+        asset,
+        destination,
+        destinationAddress,
+        moonChain,
+        moonGasLimit,
+      }) => {
+        if (!destination.wh?.name) {
+          throw new Error(
+            `Destination chain ${destination.name} does not have a wormhole name`,
+          );
+        }
+
+        const wh = new Wormhole(moonChain.isTestChain ? 'Testnet' : 'Mainnet', [
+          EvmPlatform,
+        ]);
+        const whDestinationChain = wh.getChain(destination.wh.name).config
+          .chainId;
+        const whContractAddress = (
+          isAutomatic
+            ? wh.getChain('Moonbeam').config.contracts.tokenBridgeRelayer
+            : wh.getChain('Moonbeam').config.contracts.tokenBridge
+        ) as Address;
+
+        const tokenAddressOnMoonChain = moonChain.getChainAsset(asset)
+          .address as Address | undefined;
+
+        if (!tokenAddressOnMoonChain) {
+          throw new Error(
+            `Asset ${asset.symbol} does not have a token address on chain ${moonChain.name}`,
+          );
+        }
+
+        const destinationAddress32bytes = convertAddressTo32Bytes(
+          destinationAddress,
+        ) as Address;
+        const tokenAmountOnMoonChain = asset.convertDecimals(
+          moonChain.getChainAsset(asset).decimals,
+        ).amount;
+
+        const approveTx = encodeFunctionData({
+          abi: ERC20_ABI,
+          functionName: 'approve',
+          args: [whContractAddress, tokenAmountOnMoonChain],
+        });
+        const transferTx = isAutomatic
+          ? encodeFunctionData({
+              abi: TOKEN_BRIDGE_RELAYER_ABI,
+              functionName: 'transferTokensWithRelay',
+              args: [
+                tokenAddressOnMoonChain,
+                tokenAmountOnMoonChain,
+                0,
+                whDestinationChain,
+                destinationAddress32bytes,
+                0,
+              ],
+            })
+          : encodeFunctionData({
+              abi: TOKEN_BRIDGE_ABI,
+              functionName: 'transferTokens',
+              args: [
+                tokenAddressOnMoonChain,
+                tokenAmountOnMoonChain,
+                whDestinationChain,
+                destinationAddress32bytes,
+                0n,
+                0,
+              ],
+            });
+        const batchAll = encodeFunctionData({
+          abi: BATCH_CONTRACT_ABI,
+          functionName: 'batchAll',
+          args: [
+            [tokenAddressOnMoonChain, whContractAddress],
+            [0n, 0n], // Value to send for each call
+            [approveTx, transferTx], // Call data for each call
+            [], // Gas limit for each call
+          ],
+        });
+
+        return new ExtrinsicConfig({
+          module: 'ethereumXcm',
+          func: 'transact',
+          getArgs: () => [
+            {
+              V2: {
+                gasLimit: moonGasLimit,
```

Comment:
How are you going to handle the gas estimation?

Reply:
I will get it in `mrl/src/getTransferData/getTransferData.ts` and pass it to the builder from there. Exact implementation I don't know yet, but I have some ideas.